File: buildDistribution.m
function sdt = buildDistribution(t_mean, t_std, f_mean, f_std)
% step 1: do distribution
num_stimuli = 10000;
target_strength = normrnd(t_mean, t_std, [1, num_stimuli / 2]);
foil_strength = normrnd(f_mean, f_std, [1, num_stimuli / 2]);
stimuli = [target_strength, foil_strength];

label = [true(1, num_stimuli / 2), false(1, num_stimuli / 2)];


sdt.target = target_strength; 
sdt.foil = foil_strength; 
sdt.stimuli = stimuli; 
sdt.label = label; 



end 

File: build_distribution.py
import numpy as np

def build_distribution(t_mean, t_std, f_mean, f_std):
    # Step 1: Generate distributions
    num_stimuli = 10000
    target_strength = np.random.normal(t_mean, t_std, num_stimuli // 2)
    foil_strength = np.random.normal(f_mean, f_std, num_stimuli // 2)
    
    stimuli = np.concatenate((target_strength, foil_strength))
    label = np.concatenate((np.ones(num_stimuli // 2, dtype=bool), np.zeros(num_stimuli // 2, dtype=bool)))

    sdt = {
        'target': target_strength,
        'foil': foil_strength,
        'stimuli': stimuli,
        'label': label
    }

    return sdt
